### ISSUE TITLE
Expose PDB TypeFormatter

### ISSUE THREAD

**Issue by jan-auer** [getsentry/symbolic#205]
`TypeFormatter` is internal only at the moment, but it can be useful to use publically. Also, a better place may be in the `pdb` crate.

The challenge will be to expose the name from `Function` in a way that it can be "demangled" or formatted with custom parameters. At the moment, the name must be a string, so `Name` might have to be adapted to that.

**Comment by mstange** [comment]
I just released a [pdb-addr2line crate](https://github.com/mstange/pdb-addr2line), and it also includes a [`TypeFormatter`](https://docs.rs/pdb-addr2line/0.1.0/pdb_addr2line/struct.TypeFormatter.html). It handles more cases than the one in this repository; it can print argument names and return types. It's based on the TypeDumper code from the dump_syms repo but I adapted it to make it closer to your TypeFormatter's API.

The TypeFormatter in this repo has two advantages that pdb-addr2line's TypeFormatter doesn't have yet:

 - It builds the Id index and the Type index lazily.
 - It supports cross-module imports.

I'm planning to address those shortcomings soon.

It may make sense for symbolic to switch to pdb-addr2line's TypeFormatter once these issues are addressed; doing so will give you function arguments in stack traces.

> The challenge will be to expose the name from `Function` in a way that it can be "demangled" or formatted with custom parameters. At the moment, the name must be a string, so `Name` might have to be adapted to that.

I'm not sure whether pdb-addr2line addresses this, I don't completely understand the problem. But I'm definitely open to adapting the API to work for this use case.
